sys/shell/cmds/vfs.c

```diff
@@ -744,7 +744,19 @@ static void _write_block(int fd, unsigned bs, unsigned i)
     char block[bs];
     char *buf = block;
 
-    buf += snprintf(buf, bs, "|%03u|", i);
+    int size_wanted = snprintf(buf, bs, "|%03u|", i);
```

Comment:
the return value of this is 5 `%03u` is constant width unless it failed -> bs needs to be 6 to have blocks big enough ( fit this print and a /n )

this would be more readable (to me) if instead of pointer arithmetics, adresses of array positions would be used

Reply:
Hm not entirely sure but while thinking of it, I found a neat little optimization. I moved the memset into the if-clause, please have a look.